stdlib/bytes.ml

```diff
@@ -97,19 +97,30 @@ let iter f a =
 let iteri f a =
   for i = 0 to length a - 1 do f i (unsafe_get a i) done
 
+let () = assert (3 * Sys.max_string_length <= max_int)
```

Comment:
I agree 100% with @dbuenzli here. @lefessan weren't you the one who once proposed a change of runtime representation that would allow strings up to `max_int` in length?

Also, this assertion (if we decide to keep it) should be located in `Sys` because it's part of the specification of `max_string_length`.